sapi/phpdbg/phpdbg_prompt.c

```diff
@@ -1386,7 +1386,7 @@ PHPDBG_COMMAND(dl) /* {{{ */
 			path = estrndup(param->str, param->len);
 
 			phpdbg_activate_err_buf(1);
-			if ((type = phpdbg_load_module_or_extension(&path, &name)) == NULL) {
+			if ((type = phpdbg_load_module_or_extension(&path, (const char **) &name)) == NULL) {
```

Comment:
Heh, we're nearly there! Now, changing `char *name` to `const char *name` in the variable name above should allow us to avoid casts completely if I see it right.

Reply:
> Heh, we're nearly there! Now, changing `char *name` to `const char *name` in the variable declaration above should allow us to avoid casts completely if I see it right.

ha, but we should convert `efree(name);`:

```bash
/Users/hantaohuang/codeDir/cCode/php-src/sapi/phpdbg/phpdbg_prompt.c:1391:5: warning: passing 'const char *' to parameter of type 'void *' discards qualifiers
      [-Wincompatible-pointer-types-discards-qualifiers]
                                efree(name);
```

Reply:
I think this call to `efree` is actually a bug. The name should not be freed. (It also looks like the whole `dl` command is effectively dead code right now, because it's not exposed to users.)

So I think you can simply drop this call.